src/backend/common/queries/team_query.py

```diff
@@ -94,10 +95,13 @@ def __init__(self, district_key: DistrictKey) -> None:
     def _query_async(
         self, district_key: DistrictKey
     ) -> Generator[Any, Any, List[Team]]:
+        year = int(re.match(r"\d{4,}", district_key).group(0))
         district_teams = yield DistrictTeam.query(
             DistrictTeam.district_key == ndb.Key(District, district_key)
         ).fetch_async()
         team_keys = map(lambda district_team: district_team.team, district_teams)
+        event_teams = EventTeam.query(EventTeam.year == year).fetch()
```

Comment:
I know this is a somewhat expensive query, though I'm hoping this is fine since queries are cached.